framework/wazuh/kvdbs.py

```diff
@@ -0,0 +1,402 @@
+# Copyright (C) 2015, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+from typing import List, Dict, Any, Optional
+import json
+from os import remove
+from os.path import exists
+
+from wazuh.core.results import AffectedItemsWazuhResult
+from wazuh.core.exception import WazuhException, WazuhError
+from wazuh.rbac.decorators import expose_resources
+from wazuh.core.utils import process_array, full_copy, safe_move
+
+from wazuh.core.engine import get_engine_client
+from wazuh.core.engine.utils import validate_response_or_raise
+from wazuh.core.engine.models.policies import PolicyType
+from wazuh.core.engine.models.resources import ResourceType, ResourceFormat
+
+from wazuh.core.assets import save_asset_file, generate_asset_file_path
+
+DEFAULT_KVDB_FORMAT = ResourceFormat.JSON
+ENGINE_USER_NAMESPACE = 'user'
+
+
+def _to_policy_type(policy: Optional[str]) -> Optional["PolicyType"]:
+    """Map API 'type' query param to engine PolicyType. Defaults to PRODUCTION."""
+    return PolicyType.TESTING if policy == 'testing' else PolicyType.PRODUCTION
+
+
+@expose_resources(actions=['kvdbs:read'], resources=['*:*:*'])
+async def list_kvdbs(policy_type: Optional[str] = None,
+                     ids: Optional[List[str]] = None,
```

Comment:
I believe we stopped using `Optional` typings because we can do `List[str] | None`, so we should replace these (other places in the PR also use it, ww should replace them as well)